copy/copy.go

```diff
@@ -114,28 +125,33 @@ func Image(ctx *types.SystemContext, policyContext *signature.PolicyContext, des
 	if options != nil && options.RemoveSignatures {
 		sigs = [][]byte{}
 	} else {
+		writeReport("Getting image source signatures\n")
 		s, err := src.Signatures()
 		if err != nil {
 			return fmt.Errorf("Error reading signatures: %v", err)
 		}
 		sigs = s
 	}
 	if len(sigs) != 0 {
+		writeReport("Checking if image destination supports signatures\n")
 		if err := dest.SupportsSignatures(); err != nil {
 			return fmt.Errorf("Can not copy signatures: %v", err)
 		}
 	}
 	canModifyManifest := len(sigs) == 0
 
+	writeReport("Getting image source configuration\n")
 	srcConfigInfo, err := src.ConfigInfo()
 	if err != nil {
 		return fmt.Errorf("Error parsing manifest: %v", err)
 	}
 	if srcConfigInfo.Digest != "" {
-		destConfigInfo, err := copyBlob(dest, rawSource, srcConfigInfo, false)
+		writeReport("Uploading blob %s\n", srcConfigInfo.Digest)
+		destConfigInfo, err := copyBlob(dest, rawSource, srcConfigInfo, false, options.ReportWriter)
 		if err != nil {
 			return err
 		}
+		writeReport("\n")
```

Comment:
Actually, these `'\n'` writes logically belong _into_ `copyBlob` (perhaps inside a `defer`), don’t they? The `writeReport` above in this function is properly terminated; terminating a line should pair with the code which has created the unterminated one.